src/client/sandbox/code-instrumentation/methods.js

```diff
@@ -77,23 +79,44 @@ export default class MethodCallInstrumentation extends SandboxBase {
         return args;
     }
 
+    // OPTIMIZATION: http://jsperf.com/call-apply-optimization
+    static _fastApply (owner, methName, args) {
+        var meth = owner[methName];
+
+        switch (args.length) {
+            case 1:
+                return meth.call(owner, args[0]);
+            case 2:
+                return meth.call(owner, args[0], args[1]);
+            case 3:
+                return meth.call(owner, args[0], args[1], args[2]);
+            case 4:
+                return meth.call(owner, args[0], args[1], args[2], args[3]);
+            case 5:
+                return meth.call(owner, args[0], args[1], args[2], args[3], args[4]);
+            default:
+                return meth.apply(owner, args);
+        }
+    }
+
     attach (window) {
         super.attach(window);
 
-        window[CALL_METHOD_METH_NAME] = (owner, methName, args) => {
-            if (isNullOrUndefined(owner)) {
-                MethodCallInstrumentation._error('Cannot call method \'' + methName + '\' of ' +
-                                                 inaccessibleTypeToStr(owner));
-            }
+        window[INSTRUCTION.callMethod] = (owner, methName, args) => {
+            if (isNullOrUndefined(owner))
+                MethodCallInstrumentation._error(`Cannot call method '${methName}' of ${inaccessibleTypeToStr(owner)}`);
 
             if (typeof owner[methName] !== 'function')
-                MethodCallInstrumentation._error('\'' + methName + '\' is not a function');
+                MethodCallInstrumentation._error(`'${methName}' is not a function`);
 
-            if (typeof methName !== 'string' || !this.methodWrappers.hasOwnProperty(methName))
-                return owner[methName].apply(owner, args);
+            // OPTIMIZATION: previously we've performed
+            // `this.methodWrappers.hasOwnProperty(methName)` check
+            // which is quite slow. Now we use fast RegExp check instead.
```

Comment:
OPTIMIZATION: previously we've performed the `this.methodWrappers.hasOwnProperty(methName)` check which is quite slow. Now we use the fast RegExp check instead.